crates/driver/src/tests/setup/blockchain.rs

```diff
@@ -90,7 +90,7 @@ impl Pool {
 #[derive(Debug, Clone)]
 pub struct Solution {
     pub trades: Vec<Trade>,
-    pub flashloans: Vec<Flashloan>,
+    pub flashloans: HashMap<order::Uid, Flashloan>,
```

Comment:
Is there a reason to not have the flashloan as an optional field on trade itself?

Reply:
Mostly 2 reasons:
1. the trades are enums which would result in a lot of boiler plate code for something we don't need much in the tests
2. this mimics more closely how the solution looks like in the regular code which IMO is helpful

Note that applying this suggestion to the main logic would make it harder to detect the case where the solver wants to let the driver handle the flashloans entirely.